src/Paket.Core/PaketConfigFiles/LockFile.fs

```diff
@@ -854,7 +855,7 @@ type LockFile (fileName:string, groups: Map<GroupName,LockFileGroup>) =
             for p in g.Value.NugetPackages do
                 let k = g.Key,p.Name
                 if usedPackages.ContainsKey k |> not then
-                    usedPackages.Add(k,p)
+                    usedPackages.Add(k,{p with Settings = lockGroup.Options.Settings})
```

Comment:
We we're getting the lock file's framework restrictions here. Let me know if I should use another way to get this and revert this change.